src/common/Requirements.ts

```diff
@@ -86,54 +88,64 @@ export abstract class BaseSetup implements RequirementList {
         ];
     }
     public async executeSetup(): Promise<SetupTestResult> {
-        const allPromises: Array<Promise<any>> = [];
-        this.requirements.forEach((requirement) =>
-            allPromises.push(WrappedPromise(requirement.checkFunction()))
-        );
-        const logger = this.logger;
-        const reqs = this.requirements;
-        return Promise.all(allPromises).then((results) => {
-            const testResult: SetupTestResult = {
-                hasMetAllRequirements: true,
-                tests: []
-            };
-            let count = 0;
-            results.forEach((result) => {
-                if (result.status === 'fulfilled') {
-                    testResult.tests.push({
-                        hasPassed: true,
-                        message: result.v,
-                        testResult: 'Passed'
-                    });
-                } else if (result.status === 'rejected') {
-                    testResult.hasMetAllRequirements = false;
-                    testResult.tests.push({
-                        hasPassed: false,
-                        message: result.e,
-                        testResult: 'Failed'
-                    });
-                }
-                count++;
-            });
-
-            const tree = cli.tree();
-            tree.insert('Setup');
-            testResult.tests.forEach((test) => {
-                tree.nodes.Setup.insert(
-                    `${
-                        test.hasPassed
-                            ? chalk.bold.green(test.testResult)
-                            : chalk.bold.red(test.testResult)
-                    }: ${
-                        test.hasPassed
-                            ? chalk.green(test.message)
-                            : chalk.red(test.message)
-                    }`
-                );
-            });
-            tree.display();
-            return Promise.resolve(testResult);
+        const testResult: SetupTestResult = {
+            hasMetAllRequirements: true,
+            tests: []
+        };
+
+        let totalDuration: number = 0;
+        let stepDuration: number = 0;
+        const obs = new PerformanceObserver((items) => {
+            stepDuration = items.getEntries()[0].duration / 1000;
+            performance.clearMarks();
         });
+        obs.observe({ entryTypes: ['measure'] });
+
+        for (const requirement of this.requirements) {
+            stepDuration = 0;
+            performance.mark('Start');
+            const wrappedPromise = WrappedPromise(requirement.checkFunction());
+            const result = await wrappedPromise;
```

Comment:
@trooper2013 Can you advise on this change, vs. the original [`Promise.all()` implementation above](https://github.com/forcedotcom/lwc-dev-mobile/pull/71/files#diff-9c2701a695c0d85b311d6dc97eeefd5c2ababcdaa175eb0a0189636613f5f72fL95)? I think there was a reason we did a `Promise.all()` originally, that we may lose if we do it this way.

Reply:
...and thinking it through, I think we may want to take a slightly different approach to our perf measuring implementation here. Ideally, we don't want to change the nature of our implementation simply to support performance measurements.

I think this would be a good place to make further use of `WrappedPromise`. We could further wrap the execution of the `checkFunction` Promise, in `WrappedPromise` itself, doing the metrics measurement before and after. And then the duration can just be another data field passed back in the result interface, e.g. `{ v, status: 'fulfilled', duration: duration }`.